src/Data/ArrayBuffer/DataView.js

```diff
@@ -3,43 +3,35 @@
 // module Data.ArrayBuffer.DataView
 
 
-exports.whole = function(b) {
-  return new DataView(b);
-}
-
-exports.sliceImpl = function(just, nothing, s, l, b) {
-  return ((s + l)>>>0) <= b.byteLength ? just(new DataView(b, s, l)) : nothing;
-}
-
-exports.buffer = function(v) {
-  return v.buffer;
-}
-
-exports.byteOffset = function(v) {
-  return v.byteOffset;
-}
-
-exports.byteLength = function(v) {
-  return v.byteLength;
-}
-
-exports.getterImpl = function(just, nothing, s, l, e, v, o) {
-  return function() {
-    return ((o + l)>>>0) <= v.byteLength? just(v[s].call(v,o,e)) : nothing;
-  };
-}
-
-exports.setter = function(s) {
-  return function(e) {
-    return function(v) {
-      var f = v[s];
-      return function(n) {
-        return function(o) {
-            return function() {
-            f.call(v,o,n,e);
-          };
-        };
-      };
-    };
-  };
-}
+exports.whole = function whole (b) {
+    return new DataView(b);
+};
+
+exports.remainderImpl = function remainderImpl (b,i) {
+    return new DataView(b,i);
+};
+
+exports.partImpl = function partImpl (b,i,j) {
+    return new DataView(b,i,j);
+};
+
+exports.buffer = function buffer (v) {
+    return v.buffer;
+};
+
+exports.byteOffset = function byteOffset (v) {
+    return v.byteOffset;
+};
+
+exports.byteLength = function byteLength (v) {
+    return v.byteLength;
+};
+
+exports.getterImpl = function getterImpl (s, l, e, v, o) {
+    return v[s].call(v,o,e);
+};
+
```

Comment:
The old approach (returning Maybe) is 3 times faster.

Reply:
You're sure? Is it because of the imports, or something more fundamental? I'd be happy to make a bench suite, there's some nice libs out there.

Reply:
I guess it's just that `try` is expensive.

This is what I used:

```

module Test.Benchmark where

import Prelude

import Benchmark (fn, runBench)
import Data.ArrayBuffer.DataView (getter, getter', whole)
import Data.ArrayBuffer.Typed (buffer, fromArray)
import Data.ArrayBuffer.Types (ArrayView, Float32)
import Effect (Effect)

benchmark :: Effect Unit
benchmark = runBench do
  let av :: ArrayView Float32
      av = fromArray [1.0,2.0,4.0]
      dv = whole (buffer av)
  fn "goodgetter" (\x -> getter "Float32" 1 true dv x) 0
  fn "goodgetter'" (\x -> getter' "Float32" 1 true dv x) 0
  fn "badgetter" (\x -> getter "Float32" 1 true dv x) 100
  fn "badgetter'" (\x -> getter' "Float32" 1 true dv x) 100

```


I used my fork of `purescript-benchmark` for benchmarking, there's a pending PR on that:

    "purescript-benchmark": "jacereda/purescript-benchmark-1#master"

Reply:
I'm just going to take your word for this one :) I've gotten way too sidetracked to invest in a proper benchmark suite, but I'll eventually get one fleshed out. I'd just rather have something for everyone to use for now.